draft-barnes-acme.md

```diff
@@ -1459,14 +1450,131 @@ TODO
+
+To first order, ACME assures that only authorized entities can manage certificates by requiring that a client's request for a management action be signed with an account key, and checking the authorizations attached to the account key.  The question is then whether the account recovery processes described in {{account-recovery}} can be exploited by an attacker to hijack the authorizations attached to one key and assign them to a key of the attacker's choosing.
+
+Recovery takes place in two steps:
+1. Provisioning recovery information (contact or recovery key)
+2. Using recovery information to recover an account
+
+The provisioning process needs to ensure that only the account key holder ends up with information that is useful for recovery.  The recovery process needs to assure that only the (now former) account key holder can successfully execute recovery, i.e., that this entity is the only one that can choose the new account key that receives the capabilities held by the account being recovered.
+
+MAC-based recovery only needs to assure these properties in the face of an ACME man in the middle.  MAC-based recovery only uses the ACME channel, and that channel is protected from network attackers by the use of HTTPS.
+
+In provisioning for MAC-based recovery, the recovery key is only accessible to the account private key holder, because it depends on the private ECDH key generated by that entity.  Even though an ACME MitM can observe the key agreement messages in the "recoveryKey" field of the registration transaction, the use of ECDH prevents him from learning the recovery key.
+
+Likewise, the use of HMAC in the MAC-based recovery process proves to the server that the requestor has the recovery key without exposing the recovery key.  The fact that the MAC covers the new account key means the ACME MitM can't replay it to authorize a key of his choosing.
+
+Contact-based recovery uses both the ACME channel and the contact channel.  The provisioning process is only visible to an ACME MitM, and even then, the MitM can only observe the contact information provided.  If the ACME attacker does not also have access to the contact channel, there is no risk.
+
+The security of the contact-based recovery process is entirely dependent on the security of the contact channel.  The details of this will depend on the specific out-of-band technique used by the server.  For example:
+
+* If the server requires a user to click a link in a message sent to a contact address, then the contact channel will need to provide confidentiality and authenticity (of the destination) in order to be secure.  Otherwise, a passive attacker could see the link and click it first, or an active attacker could redirect the message.
+* If the server requires a user to respond to a message sent to a contact address, then the contact channel will need to provide source and destination authenticity.  Otherwise an active attacker can spoof a message from the contact address.
+
```

Comment:
This seems backwards. We usually expect that it provides confidentiality for the message sent *to* the contact address.